ntp.go

```diff
@@ -139,12 +144,23 @@ type Response struct {
 	Leap           LeapIndicator // server's leap second indicator; see RFC 5905
 }
 
+type QOption struct {
+	Timeout time.Duration // defaults to defaultTimeout
+	Version int           // NTP protocol version, defaults to 4
+	Port    int           // NTP Server port for UDPAddr.Port, defaults to 123
+	IpTTL   int           // IP TTL to use for outgoing UDP packets
+}
+
 // Query returns the current time from the remote server host using the
 // requested version of the NTP protocol. It also returns additional
 // information about the exchanged time information. The version may be 2, 3,
 // or 4; although 4 is most typically used.
 func Query(host string, version int) (*Response, error) {
-	m, err := getTime(host, version)
+	return QueryEx(host, QOption{Version: version})
+}
+
+func QueryEx(host string, opt QOption) (*Response, error) {
```

Comment:
Let's call this `QueryWithOptions` instead.  I'm not a big fan of `Ex` APIs when they can be avoided.  This function will also need a comment for the documentation.